examples/material_gallery/lib/demo/buttons_demo.dart

```diff
@@ -3,214 +3,267 @@
+    title: new Text("Demo")
+  ),
+  floatingActionButton: new FloatingActionButton(
+    child: new Icon(icon: Icons.add)
+  )
+);""";
 
 class ButtonsDemo extends StatefulWidget {
   @override
   _ButtonsDemoState createState() => new _ButtonsDemoState();
 }
 
 class _ButtonsDemoState extends State<ButtonsDemo> {
-  List<_ButtonDemo> _demos;
+  List<ComponentDemoTabData> _demos;
 
   @override
   void initState() {
     super.initState();
-    _demos = <_ButtonDemo>[
-      new _ButtonDemo(title: 'FLOATING', text: _floatingText, builder: buildFloatingButton),
-      new _ButtonDemo(title: 'RAISED', text: _raisedText, builder: buildRaisedButton),
-      new _ButtonDemo(title: 'FLAT', text: _flatText, builder: buildFlatButton),
-      new _ButtonDemo(title: 'DROPDOWN', text: _dropdownText, builder: buildDropdownButton)
+    _demos = <ComponentDemoTabData>[
+      new ComponentDemoTabData(
+        tabName: 'RAISED',
+        description: _raisedText,
+        builder: buildRaisedButton,
+        exampleCode: _raisedCode
+      ),
+      new ComponentDemoTabData(
+        tabName: 'FLAT',
+        description: _flatText,
+        builder: buildFlatButton,
+        exampleCode: _flatCode
+      ),
+      new ComponentDemoTabData(
+        tabName: 'DROPDOWN',
+        description: _dropdownText,
+        builder: buildDropdownButton,
+        exampleCode:
+        _dropdownCode
+      ),
+      new ComponentDemoTabData(
+        tabName: 'ICON',
+        description: _iconText,
+        builder: buildIconButton,
+        exampleCode: _iconCode
+      ),
+      new ComponentDemoTabData(
+        tabName: 'ACTION',
+        description: _actionText,
+        builder: buildActionButton,
+        exampleCode: _actionCode
+      ),
     ];
   }
 
-  Widget buildFloatingButton(BuildContext context) {
-    return new SizedBox(
-      height: 128.0,
-      child: new Center(
-        child: new FloatingActionButton(
-          tooltip: 'Open FAB demos',
-          child: new Icon(icon: Icons.add),
-          onPressed: () {
-            Navigator.push(context, new MaterialPageRoute<Null>(
-              builder: (BuildContext context) => new TabsFabDemo()
-            ));
-          }
-        )
+  Widget buildActionButton(BuildContext context) {
+    return new Align(
+      alignment: new FractionalOffset(0.5, 0.4),
+      child: new FloatingActionButton(
+        child: new Icon(icon: Icons.add),
+        onPressed: () {
+          // Perform some action
+        }
       )
     );
   }
 
   Widget buildRaisedButton(BuildContext context) {
-    return new Container(
-      margin: const EdgeInsets.symmetric(vertical: 16.0),
-      child: new Column(
+    return new Align(
+      alignment: new FractionalOffset(0.5, 0.4),
+      child: new Row(
+        mainAxisAlignment: MainAxisAlignment.collapse,
         children: <Widget>[
           new RaisedButton(
-            child: new Text('LAUNCH DEMO'),
+            child: new Text("RAISED BUTTON"),
```

Comment:
Yes (except the `'''` multiline strings): https://flutter.io/style-guide/#prefer-single-quotes-for-strings